pkg/controller/gkenetworkparamset/gkenetworkparamset_controller.go

```diff
@@ -94,9 +100,25 @@ func (c *Controller) Run(numWorkers int, stopCh <-chan struct{}, controllerManag
 		},
 	})
 
-	if !cache.WaitForNamedCacheSync("gkenetworkparamset", stopCh, c.gkeNetworkParamsInformer.HasSynced) {
+	c.networkInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			network := obj.(*networkv1.Network)
+			if network.Spec.ParametersRef != nil && network.Spec.ParametersRef.Kind == gnpKind {
```

Comment:
can this happen?

Reply:
I'm not 100% sure what you mean. If youre asking if this if condition will ever be true, then yes.

Reply:
I just wanted to know if there is validation on the network.Parameters that guarantees this condition

Reply:
network.Spec.ParametersRef has 3 cases.

1. nil (when the network resource is backed without a managed cloud environment like gcp)
2. not nil, but points to a different type of params object (could eventually be something like awsParams)
3. not nil and points to a GNP object (I want to process to these)

Reply:
those comments on the code are very helpful for future us or reviewers ;)

Reply:
added